openmrs-client/src/main/java/org/openmrs/mobile/api/retrofit/PatientApi.java

```diff
@@ -194,14 +194,21 @@ public void onFailure(Call<PatientPhoto> call, Throwable t) {
     public void updatePatient(final Patient patient, @Nullable final DefaultResponseCallbackListener callbackListener) {
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(OpenMRS.getInstance());
         Boolean syncstate = prefs.getBoolean("sync", true);
-        new PatientDAO().updatePatient(patient.getId(), patient);
         if (syncstate) {
             RestApi restApi = RestServiceBuilder.createService(RestApi.class);
             Call<Patient> call = restApi.updatePatient(patient, patient.getUuid(), "full");
             call.enqueue(new Callback<Patient>() {
                 @Override
                 public void onResponse(Call<Patient> call, Response<Patient> response) {
                     if (response.isSuccessful()) {
+                        patient.getPerson().setUuid(patient.getUuid());
+
+                        if (patient.getPerson().getPhoto() != null) {
+                            uploadPatientPhoto(patient);
+                        }
+
+                        new PatientDAO().updatePatient(patient.getId(), patient);
```

Comment:
I moved it here because I think we should save user to the DB only after successful response. Without that app might crash when trying to update patient